src/javax/microedition/io/file/FileConnection.java

```diff
@@ -26,65 +27,76 @@
 
 public interface FileConnection extends StreamConnection 
 {
-    public long availableSize();
+    long availableSize() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public boolean canRead();
+    boolean canRead() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public boolean canWrite();
+    boolean canWrite() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public void create();
+    void create() throws IOException, SecurityException, IllegalModeException;
 
-    public void delete();
+    void delete() throws IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public long directorySize(boolean includeSubDirs);
+    long directorySize(boolean includeSubDirs) throws IOException, SecurityException, IllegalModeException, 
+    ConnectionClosedException;
 
-    public boolean exists();
+    boolean exists() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public long fileSize();
+    long fileSize() throws IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public String getName();
+    String getName();
 
-    public String getPath();
+    String getPath();
 
-    public String getURL();
+    String getURL();
 
-    public boolean isDirectory();
+    boolean isDirectory() throws IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public boolean isHidden();
+    boolean isHidden() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public boolean isOpen();
+    boolean isOpen();
     
-    public long lastModified();
+    long lastModified() throws SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public Enumeration list();
+    Enumeration list() throws IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public Enumeration list(String filter, boolean includeHidden);
+    Enumeration list(String filter, boolean includeHidden) throws NullPointerException, IllegalArgumentException, 
+    IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public void mkdir();
+    void mkdir() throws IOException, SecurityException, IllegalModeException, ConnectionClosedException;
 
-    public DataInputStream openDataInputStream();
+    @Override /* Throws IOException in the docs, but not on the overriden class */
+    DataInputStream openDataInputStream() throws SecurityException, IllegalModeException;
 
-    public DataOutputStream openDataOutputStream();
+    @Override /* Throws IOException in the docs, but not on the overriden class */
```

Comment:
" Throws IOException in the docs, but not on the overriden class"  What does this mean? What other class?  Remember that this is an API.  That means it is entirely separate from any specific application that uses it.

Reply:
Oh, it means that this DataOutputStream method is described as throwing an IOException in the javadocs. However, our current implementation (the compiler at least complained about not overriding a function with the same name as that) doesn't. I'll clean it up later when i do a second pass on all those files.